core/arch/arm/kernel/boot.c

```diff
@@ -188,6 +188,26 @@ static void init_vfp_nsec(void)
 }
 #endif
 
+/*
+ * Check for supported Crypto Extensions (ARMv8 aarch32/aarch64)
+ * In case one of instructions is not supported false is returned.
+ */
+static bool check_cpuid_ce(void)
+{
+	uint32_t isar5 = read_id_isar5();
```

Comment:
Shouldn't we use `ID_AA64ISAR0_EL1` when in AArch64? I guess we could even take the different configuration options
`CFG_CRYPTO_AES_ARM_CE`, `CFG_CRYPTO_SHA1_ARM_CE`, `CFG_CRYPTO_SHA256_ARM_CE`, `CFG_CRYPTO_SHA512_ARM_CE`, `CFG_CRYPTO_SHA3_ARM_CE`, `CFG_CRYPTO_SM3_ARM_CE`, `CFG_CRYPTO_SM4_ARM_CE`, and `CFG_CRYPTO_SM4_ARM_AESE` into consideration when testing in this function.

Reply:
> Shouldn't we use ID_AA64ISAR0_EL1 when in AArch64

on one hand it would be correct, on the other hand I wanted to keep it as simple as possible and address the issue described in https://github.com/OP-TEE/optee_os/issues/6631, without adding any logic for differentiation of versions of ARMv8 CE. Btw, this is neither done in makescripts,  as enabling `CFG_CRYPTO_WITH_CE` implies `CFG_CRYPTO_*_ARM_CE` options (unless disabled explicitly), expecting the code to be running at least on Armv8.2-A core.

> I guess we could even take the different configuration options CFG_CRYPTO_AES_ARM_CE, CFG_CRYPTO_SHA1_ARM_CE, CFG_CRYPTO_SHA256_ARM_CE, CFG_CRYPTO_SHA512_ARM_CE, CFG_CRYPTO_SHA3_ARM_CE, CFG_CRYPTO_SM3_ARM_CE, CFG_CRYPTO_SM4_ARM_CE, and CFG_CRYPTO_SM4_ARM_AESE into consideration when testing in this function.

I agree, and ideally that would make more sense if we included both software and arm-ce implementations of these algos so the proper one can be chosen in runtime.

Reply:
@jenswi-linaro added new implementation